README.md

```diff
@@ -5,23 +5,181 @@ Record any website using our recorder, or deploy our service to manage it for yo
+file_output:
+    local: true
+```
+
+Next, create a `request.json`:
+```json
+{
+  "url": "https://www.youtube.com/watch?v=dQw4w9WgXcQ",
+  "filepath": "/out/demo.mp4"
+}
+```
+
+Start the recording:
+```shell
+mkdir -p ~/livekit/recordings
+
+docker run --rm --name demo \
+    -e LIVEKIT_RECORDER_CONFIG="$(cat config.yaml)" \
+    -e RECORDING_REQUEST="$(cat basic.json)" \
+    -v ~/livekit/recordings:/out \
+    livekit/livekit-recorder
+```
+
+Then, to stop the recording:
+```shell
+docker stop demo
+```
+
+You should find a `~/livekit/recordings/demo.mp4`.
+
+## Uploading to S3
+
+Update your config.yaml and replace
+```yaml
+file_output:
+    local: true
+```
+with
+```yaml
+file_output:
+    s3:
+        access_key: <s3-access-key>
+        secret: <s3-secret>
+        region: <s3-region>
+        bucket: <s3-bucket>
+```
+
+Create a `s3.json`:
+```json
+{
+    "url": "https://www.youtube.com/watch?v=BHACKCNDMW8",
+    "filepath": "path/filename.mp4",
+    "options": {
+        "preset": "HD_60"
+    }
+}
+```
+This time, we've added the `HD_60` preset. This will record at 1280x720, 60fps (the default is 1920x1080, 30fps).
+You can find the other presets and options [below](#presets).
+
+```shell
+docker run --rm --name demo2 \
+    -e LIVEKIT_RECORDER_CONFIG="$(cat config.yaml)" \
+    -e RECORDING_REQUEST="$(cat s3.json)" \
+    livekit/livekit-recorder
+```
+
+```shell
+docker stop demo2
+```
+
+After the recording is stopped, the file will be uploaded to your S3 bucket.
+
+## Rtmp Output
+
+Create a `rtmp.json` (if you have a Twitch account you can fill in your stream key, otherwise replace the rtmp url with your provider):
+```json
+{
+    "url": "https://www.youtube.com/watch?v=aW2LvQUcwqc",
+    "rtmp": {
+        "urls": ["rtmp://live.twitch.tv/app/<stream-key>"]
+    },
+    "options": {
+        "width": "1280",
+        "height": "720",
+        "video_bitrate": 2048
+    }
+}
+```
+This time, we've set custom options to output 720p with a lower bitrate (2048 kbps - the default is 3000 kpbs). If you have good bandwidth at home, try using `preset: FULL_HD_60` instead for a high quality stream.
+
+Start the stream:
+```shell
+docker run --rm --name demo3 \
+    -e LIVEKIT_RECORDER_CONFIG="$(cat config.yaml)" \
+    -e RECORDING_REQUEST="$(cat rtmp.json)" \
+    livekit/livekit-recorder
+```
+Note: with Twitch, it will take about 25 seconds for them to process before they begin showing the stream. 
+May be different with other providers.
+
+Stop the stream:
+```shell
+docker stop demo3
+```
+
+## Recording rooms
+
+So far, we've only been recording YouTube videos. Let's look at how to record a LiveKit room.
+
+To start, you'll need to have your LiveKit server [deployed](https://docs.livekit.io/deploy) with valid SSL certificates,
+using the same `api_key` and `api_secret` as the ones in your `config.yaml`. 
+The recorder will not work with a local LiveKit server, as the server needs to be reachable by our hosted templates.
```

Comment:
I don't think this is accurate. the only requirement is that user is able to reach the hosted template. web app executes within the browser (i.e. docker instance). The only requirement should be that the docker environment that runs the recorder could reach both `recorder.livekit.io` *and* the livekit server